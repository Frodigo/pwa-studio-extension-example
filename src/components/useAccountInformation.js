import { useUserContext } from '@magento/peregrine/lib/context/user';

/**
 * useAccountInformation hook which provides data for AccountInformation component
 * @returns {{currentUser: {id, email, firstname, lastname, is_subscribed}, isSignedIn: boolean}}
 */
export const useAccountInformation = () => {
  const [{ currentUser, isSignedIn }] = useUserContext();

  return {
    currentUser,
    isSignedIn
  };
};
