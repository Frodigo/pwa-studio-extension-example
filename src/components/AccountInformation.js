import React from 'react';
import { useAccountInformation } from './useAccountInformation';
import { Redirect } from '@magento/venia-drivers';

const AccountInformation = () => {
  const { currentUser, isSignedIn } = useAccountInformation();

  if (!isSignedIn) {
    return <Redirect to="/" />;
  }

  return (
    <div>
      <h1>Account information</h1>

      <ul>
        <li>
          <strong>First name: </strong>
          <span>{currentUser.firstname}</span>
        </li>
        <li>
          <strong>Last name: </strong>
          <span>{currentUser.lastname}</span>
        </li>
      </ul>
    </div>
  );
};

export default AccountInformation;
