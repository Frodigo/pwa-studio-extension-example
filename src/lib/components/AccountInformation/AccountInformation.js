import React from 'react';
import { useAccountInformation } from '../../talons/AccountInformation/useAccountInformation';
import { Redirect } from '@magento/venia-drivers';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from './accountInformation.css';

const AccountInformation = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const { currentUser, isSignedIn } = useAccountInformation();

  if (!isSignedIn) {
    return <Redirect to="/" />;
  }

  return (
    <div className={classes.root}>
      <h1 className={classes.title}>Account information</h1>

      <ul className={classes.list}>
        <li>
          <strong>First name: </strong>
          <span>{currentUser.firstname}</span>
        </li>
        <li>
          <strong>Last name: </strong>
          <span>{currentUser.lastname}</span>
        </li>
      </ul>
    </div>
  );
};

export default AccountInformation;
