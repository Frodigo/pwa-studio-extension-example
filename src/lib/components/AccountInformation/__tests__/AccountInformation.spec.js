import React from 'react';
import { default as createTestInstance } from '@magento/peregrine/lib/util/createTestInstance';
import { useAccountInformation } from '../../../talons/AccountInformation/useAccountInformation';
import AccountInformation from '../AccountInformation';

jest.mock(
    '@marcinkwiatkowski/customer-menu/src/lib/talons/AccountInformation/useAccountInformation'
);
jest.mock('@magento/venia-ui/lib/classify');
jest.mock('@magento/peregrine/lib/context/user', () => {
    const userState = {
        isGettingDetails: false,
        getDetailsError: null
    };
    const userApi = {
        getUserDetails: jest.fn(),
        setToken: jest.fn(),
        signIn: jest.fn()
    };
    const useUserContext = jest.fn(() => [userState, userApi]);

    return { useUserContext };
});

jest.mock('@magento/venia-ui/lib/drivers', () => ({
    Redirect: props => <mock-Redirect {...props} />
}));

test('Redirects when not authenticated', () => {
    useAccountInformation.mockReturnValue({
        isSignedIn: false,
        currentUser: null
    });

    const tree = createTestInstance(<AccountInformation />);
    expect(tree.toJSON()).toMatchSnapshot();
});

test('Display page when user is signed in', () => {
    useAccountInformation.mockReturnValue({
        isSignedIn: true,
        currentUser: {
            firstname: 'Marcin',
            lastname: 'Kwiatkowski'
        }
    });

    const tree = createTestInstance(<AccountInformation />);
    expect(tree.toJSON()).toMatchSnapshot();
});
