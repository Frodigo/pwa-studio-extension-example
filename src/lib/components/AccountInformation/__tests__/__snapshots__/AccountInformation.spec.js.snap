// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display page when user is signed in 1`] = `
<div
  className="root"
>
  <h1
    className="title"
  >
    Account information
  </h1>
  <ul
    className="list"
  >
    <li>
      <strong>
        First name: 
      </strong>
      <span>
        Marcin
      </span>
    </li>
    <li>
      <strong>
        Last name: 
      </strong>
      <span>
        Kwiatkowski
      </span>
    </li>
  </ul>
</div>
`;

exports[`Redirects when not authenticated 1`] = `
<mock-Redirect
  to="/"
/>
`;
